### PR TITLE
#22 | Implement Cart service and #27 | Implement Orders service

### PR DESCRIPTION
**Pull request by SavinaMih**

### CODE REVIEW

**Review by MartsTech — CHANGES_REQUESTED** [review]
Why are all files related to the cart and orders service in the root of the project? They should be seperated in their own folders called Cart and Orders in the Services folder. Also some of the folders are called Cart and ShoppingCart, OrderService and OrderingService. This is not consistance. Look at the Bookmarks and Catalog service for an example how to structure the code correctly.

I see you have moved code from the already build services, but please do not leave namespaces form them. I can see several files that have bookmarks and catalog in them, which is messy.

For some reason I saw login for creating products in ShoppingCart.Persistence. Why? The Catalog Service has already build this. Take a loot at the Bookmarks service. It also relies on the products, but does not implement them on its own. The cart service for now will be very similar to the bookmarks, so you can just copy it, rename all files correctly and change the data you store.

Aeon is using **.NET 7**. This could lead to problems in production. Please update it.

You are using some form of clean arhitecture, but it is not the some the already established. For example OrderingService.Core has Controllers, but where is the Orders.Api? It is probably a fine way of doing things, but still.

Did you test your services. Are they using the same configurations as the others. The database settings, swagger, cors and event bus?

On final notes, the code is very messy right now and I cannot do a proper review on the functionality of the code. Just ask @bdbelchev if you need help with understanding how the current services work and copy the style of the services he has done.

File: Aeon.sln
Comment (bdbelchev) [inline Aeon.sln:34]: The .sln still points to the incorrect directories and the IDE can't load the projects. Fortunately, we can fix this fairly easily.

File: Item.cs
Comment (bdbelchev) [inline Item.cs:10]: I'm quite sure that it's more appropriate for names to be a string. Also, the file is left over in the root directory of the project.

File: src/Services/OrderService/OrderService.Application/OrderList/CreateList/CreateListCommand.cs
Comment (bdbelchev) [inline src/Services/OrderService/OrderService.Application/OrderList/CreateList/CreateListCommand.cs:1]: We should not use namespaces and classes from other services, as every single one should be independent and this creates a reliance and may lead to errors.

File: src/Services/OrderService/OrderService.Application/OrderList/CreateList/CreateListInput.cs
Comment (bdbelchev) [inline src/Services/OrderService/OrderService.Application/OrderList/CreateList/CreateListInput.cs:1]: Namespaces should correspond to the directory structure. And, ideally, not to other projects :)

File: src/Services/OrderService/OrderService.Application/OrderList/CreateList/CreateListInputValidator.cs
Comment (bdbelchev) [inline src/Services/OrderService/OrderService.Application/OrderList/CreateList/CreateListInputValidator.cs:3]: See my last comment.

File: src/Services/OrderService/OrderService.Application/OrderList/GetList/GetListByIdQuery.cs
Comment (bdbelchev) [inline src/Services/OrderService/OrderService.Application/OrderList/GetList/GetListByIdQuery.cs:1]: See comment from before. Ideally, every single service should not know of the existence of other services, and definitely not depend on them.

File: src/Services/OrderService/OrderService.Application/OrderService.Application.csproj
Comment (bdbelchev) [inline src/Services/OrderService/OrderService.Application/OrderService.Application.csproj:5]: I suggest we all use the latest stable version of .NET, which is 7.

File: src/Services/OrderService/OrderService.Application/Orders/AddOrder/AddOrderCommand.cs
Comment (bdbelchev) [inline src/Services/OrderService/OrderService.Application/Orders/AddOrder/AddOrderCommand.cs:3]: See previous comments.

File: src/Services/OrderService/OrderService.Application/obj/Debug/net6.0/OrderService.Application.AssemblyInfo.cs
Comment (bdbelchev) [inline src/Services/OrderService/OrderService.Application/obj/Debug/net6.0/OrderService.Application.AssemblyInfo.cs:1]: Seems that some folders bypassed the .gitignore... You can safely delete bin/obj/debug and some other folders that are generated by your IDE from the repo.

File: src/Services/OrderService/OrderService.Domain/OrderService.Domain.csproj
Comment (bdbelchev) [inline src/Services/OrderService/OrderService.Domain/OrderService.Domain.csproj:5]: See previous comments on .NET.

File: src/Services/OrderService/OrderService.Persistence/OrderLists/OrderListConfiguration.cs
Comment (bdbelchev) [inline src/Services/OrderService/OrderService.Persistence/OrderLists/OrderListConfiguration.cs:5]: See previous comments on namespaces.

**Review by bdbelchev — CHANGES_REQUESTED** [review]
I highlighted some issues in the code (most of which were already highlighted by @MartsTech), we should focus on fixing namespaces, incorrect references, updating to .NET 7 and removing the temporary directories, among other corrections. 
I also noticed some redundant logic related to products in some of the services. 

Unfortunately, I won't have enough time to conduct a full code review and test before the project defense, which is in 30 minutes, and I haven't looked deeply into the code, but overall I think we could wait until after today to fix and merge this branch, given the lack of time.

Thank you for your effort :)

**Pull request closed without merging**
